### ISSUE TITLE
TDD 1.3 not a trusted app for both Windows and Apple

### ISSUE THREAD

**Issue by lmcthbe** [OWASP/threat-dragon-desktop#43]
Threat Dragon Desktop 1.3 is not recognized as a thrusted application on both Windows and Apple.

Enterprise usually does not allow non thrusted application to be installed on managed computer. TDD 1.3 can still be installed on Windows as there is still a possibility to perform the download/installation however it is completely blocked on Mac OS preventing any installation.

**Comment by jgadsden** [comment]
Hello @lmcthbe - agreed that we can not sign the Windows nor MacOS applications at the moment. We did have a key pair for the Windows application but that is tied to @mike-goodwin 's computer.

For MacOS can you try these instructions for running the application?
https://github.com/OWASP/threat-dragon-desktop#running-on-macos

It works for me both on MacOS and Windows, and I agree it is not ideal.

**Comment by jgadsden** [comment]
Hello @lmcthbe - we use the same principle as we use for ZAP. Can you run ZAP OK? It must be part of the security settings for this MacOS. I am running 10.15.6 if that is relevant (not sure)

**Comment by lmcthbe** [comment]
@jgadsden  Yes. Same issue with 1.2. My work colleague got new MBP with MDM (Mobile Device Management) in place restricting to install non signed app (to enforce enterprise security policy)
I have also an MBP but no MDM in place yet and of course I was able to install TDD, ZAP however by end of September I will be onboarded in MDM too :-(

**Comment by jgadsden** [comment]
Hello @lmcthbe - thanks for the confirmation and investigation into the consequences of MDM on the MacOS platform.

This sounds like a severe problem, not only for us but also flagship OWASP projects like ZAP. I will have a look to see if ZAP is aware of this, and what plans they have.

**Comment by jgadsden** [comment]
@lmcthbe could you try one last thing for me? Could your colleague try navigating to Applications, and rather than double clicking if they could be sure to control+click (in effect a right click) and then select Open.
Double-click will give no option but to bin the app or cancel, but right-click should give the extra third option?
Thanks, Jon

**Comment by lmcthbe** [comment]
Will ask. Thanks

**Comment by lmcthbe** [comment]
After control-click and Open, it just gives two options : Move to Trash or Cancel

![Screen Shot 2020-09-14 at 10 50 22 AM](https://user-images.githubusercontent.com/1301592/93137109-f98f1b00-f6aa-11ea-819e-ff2fd4d88321.png)

**Comment by jgadsden** [comment]
OK, thanks for doing this @lmcthbe - this is very much a problem as your colleague (as you said) is unable to use TD. If this is the same for ZAP then we may be able to get OWASP / ZAP to take action. Threat Dragon is only an incubator project; nowhere near ZAP's flagship status ... yet :-)

I have added a comment to this effect to an existing issue in zaproxy:
https://github.com/zaproxy/zaproxy/issues/6162

**Comment by jgadsden** [comment]
I have made this the focus of version 1.4 release, as this seems to be a blocker and will also be useful for ZAP

**Comment by jgadsden** [comment]
Apple Developer Distribution Certificate:

https://www.codementor.io/@nitingohel/apple-developer-distribution-certificate-du107wwff

**Comment by jgadsden** [comment]
@koberbe has reported signing issue for the Linux snap installer, so this is becoming much more of a problem

**Comment by jgadsden** [comment]
Similar issue for Amass, using precompiled binaries:
https://github.com/OWASP/Amass/releases

**Comment by jgadsden** [comment]
Hello @lmcthbe - having looked into this and gone down many dead-ends (or blind alleys? not sure of the term) I came across a solution when installing Portswigger's BurpSuite. Could you try this when you have time please?

> When you try to run the installer on MacOS, an error message might appear saying that the installer "can't be opened because Apple cannot check it for malicious software". In order to bypass this warning, you need to manually confirm that you trust the installer and want to run it.
	.	Close the error message popup.
	.	In MacOS, go to "System Preferences" > "Security & Privacy".
	.	At the bottom of the dialog, you will see a message saying that the OWASP-Threat-Dragon was blocked. Next to it click "Open anyway".
	.	The initial error message will pop up again, but this time you have the option to click "Open" to run the installer despite the warning
	.	This only has to be done once, after doing this Threat Dragon can be opened as normal 

I am going to update the documentation to describe this method. I have tried it out and it worked for me for the version 1.3.1 dmg  download of Threat Dragon

**Comment by jgadsden** [comment]
@storkinsj  has reported this again in 'Signed Bundles' OWASP/threat-dragon#52

Needs to be done as a high priority.  The company I work for has this week given me an Apple developer licence ... so I was planning on releasing version 1.3.2 later this month with some bug fixes and properly signed downloads.

**Comment by jgadsden** [comment]
Transferred to threat dragon repo
